src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -53,9 +55,35 @@ class ActionPlanService(
     return actionPlanRepository.save(draftActionPlan)
   }
 
+  fun submitDraftActionPlan(id: UUID, submittedByUser: AuthUser): ActionPlan {
+
+    val draftActionPlan = getDraftActionPlanAndThrowExceptionIfNotFound(id)
+    updateDraftActionPlanAsSubmitted(draftActionPlan, submittedByUser)
+    val savedDraftActionPlan = actionPlanRepository.save(draftActionPlan)
+    actionPlanEventPublisher.actionPlanSubmitEvent(savedDraftActionPlan)
+
+    return savedDraftActionPlan
+  }
+
+  fun getActionPlan(id: UUID): ActionPlan {
+    return actionPlanRepository.findById(id).orElseThrow {
+      throw EntityNotFoundException("action plan not found [id=$id]")
+    }
+  }
+
   private fun updateDraftActivityPlan(draftActionPlan: ActionPlan, update: ActionPlan) {
     update.numberOfSessions?.let {
       draftActionPlan.numberOfSessions = it
     }
   }
+
+  private fun getDraftActionPlanAndThrowExceptionIfNotFound(id: UUID): ActionPlan {
```

Comment:
can we use `orElseThrow` here too? would be good to keep these methods consistent

Reply:
will do

Reply:
Done